src/pages/ResourceBatchUpdatePage.tsx

```diff
@@ -0,0 +1,316 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Notification,
+  Table,
+  Pagination,
+  RadioButton,
+  SelectionGroup,
+  IconTrash,
+} from 'hds-react';
+import {
+  PrimaryButton,
+  SecondaryButton,
+  SupplementaryButton,
+} from '../components/button/Button';
+import { useAppContext } from '../App-context';
+import api from '../common/utils/api/api';
+import { Language, Resource } from '../common/lib/types';
+import sessionStorage from '../common/utils/storage/sessionStorage';
+import './ResourceBatchUpdatePage.scss';
+import { TargetResourcesProps } from '../components/resource-opening-hours/ResourcePeriodsCopyFieldset';
+
+export type ResourceBatchUpdatePageProps = {
+  mainResourceId: string;
+  targetResourcesString?: string;
+};
+
+// for some reason origins are not part of Resource type, this need to be fixed in API
+type ResourceWithOrigins = Resource & {
+  origins: {
+    origin_id: string;
+    data_source: {
+      id: string;
+      name: {
+        fi: string;
+        sv: string;
+        en: string;
+      };
+    };
+  }[];
+};
+
+const ResourceBatchUpdatePage = ({
+  mainResourceId,
+  targetResourcesString,
+}: ResourceBatchUpdatePageProps): JSX.Element => {
+  const { language: contextLanguage } = useAppContext();
+  const language = contextLanguage || Language.FI;
+  const [resource, setResource] = useState<Resource | undefined>(undefined);
+  const [error, setError] = useState<Error | undefined>(undefined);
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [pageIndex, setPageIndex] = useState(0);
+  const [selectedRadioItem, setSelectedRadioItem] = useState('copy');
+  const [targetResourceData, setTargetResourceData] = useState<
+    TargetResourcesProps | undefined
+  >(undefined);
+  const targetResourcesStorageKey = 'targetResources';
```

Comment:
For better readability, could we move this and `language` above closer to `cols` and `rows` further down? I would nicely group constants together 🙂